File: Panacea/py3/pan.py
# panacea soln in py3 for code eval by steven a dunn

import sys

f = open(sys.argv[1], 'r')
for line in f:
	vir, ant = line.strip().split(" | ")
	vir = [int(x, 16) for x in vir.split()]
	ant = [int(x, 2) for x in ant.split()]
	vir_sum, ant_sum = sum(vir), sum(ant)
	
	if ant_sum >= vir_sum:
		print ("True")
	else:
		print ("False")
f.close()

File: Panacea/c/pan.c
// panacea soln in c for code eval by steven a dunn

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main (int argc, char* const argv[])
{
	FILE *fp = fopen(argv[1], "r");
	if (fp)
	{
		char line[1024];
		while (fgets(line, sizeof(line), fp))
		{
			char *token = strtok(line, "|");
			char* vir = token;

			token = strtok(NULL, "|");
			char* ant = token;

			int vir_sum = 0;
			token = strtok(vir, " ");
			while (token)
			{
				vir_sum += strtol(token, NULL, 16);
				token = strtok(NULL, " ");
			}

			int ant_sum = 0;
			token = strtok(ant, " ");
			while (token)
			{
				ant_sum += strtol(token, NULL, 2);
				token = strtok(NULL, " ");
			}

			if (vir_sum <= ant_sum)
				puts("True");
			else
				puts("False");
		}
		fclose(fp);
	}
	return 0;
}